addons/osio-addon/src/main/java/io/fabric8/launcher/osio/identity/OsioIdentityProvider.java

```diff
@@ -19,28 +23,33 @@
 @Application(OSIO)
 public class OsioIdentityProvider implements IdentityProvider {
 
-    private static final String GITHUB_SERVICENAME = "https://github.com";
 
     private final TokenIdentity authorization;
-    private final OsioAuthClient authClient;
 
     @Inject
-    public OsioIdentityProvider(final TokenIdentity authorization, final OsioAuthClient osioAuthClient) {
+    public OsioIdentityProvider(final TokenIdentity authorization) {
         this.authorization = requireNonNull(authorization, "authorization must be specified.");
-        this.authClient = requireNonNull(osioAuthClient, "osioAuthClient must be specified.");
     }
 
     @Override
     public Optional<Identity> getIdentity(final String service) {
+        return getIdentity(authorization, service);
+    }
+
+    @Override
+    public Map<String, Optional<Identity>> getIdentities(final Set<String> services) {
+        final TokenIdentity immutableAuthorization = ImmutableTokenIdentity.copyOf(authorization);
+        return services.stream().parallel()
```

Comment:
Using `parallel` in containers is a potential issue. If you can remember when using the `Future`s, you were obliged to use the app server's executor service, this is because you can't  let Java use it default thread pools. But here, using `parallel()` you're doing that under the hood. So, not good.

So either you would pass an `ExecutorService` to this method, or like I suggested in my other comment, you remove the parallelism here and move it up the call chain far enough until you reach a point where you can easily retrieve the app server's executor service.

Reply:
You can `@Inject ExecutorService` in the constructor in this case